rsmtool/utils/commandline.py

```diff
@@ -230,117 +253,620 @@ def existing_configuration_file(string):
+        ----------
+        allow_empty : bool, optional
+            If ``True``, it will allow the user to also just press
+            enter (i.e., input a blank string), in addition to "true"
+            or "false".
+            Defaults to ``False``.
+
+        Returns
+        -------
+        validator : prompt_toolkit.validation.Validator
+            A ``Validator`` instance that ensures that the user
+            input for the field is "true" / "false", or possibly
+            the empty string, if ``allow_empty`` is ``True``.
+        """
+        correct_choices = ["true", "false"]
+        if allow_empty:
+            correct_choices.append("")
+        validator = Validator.from_callable(lambda answer: answer in correct_choices,
+                                            error_message="invalid answer")
+        return validator
+
+    def _make_choice_validator(self, choices):
+        """
+        Private method that creates a validator for a field
+        with ``data_type`` of "choice".
+
+        Parameters
+        ----------
+        choices : list
+            List of possible values for the field.
+
+        Returns
+        -------
+        validator : prompt_toolkit.validation.Validator
+            A ``Validator`` instance that ensures that the user
+            input for the field is one of the possible choices.
+        """
+        validator = Validator.from_callable(lambda choice: choice in choices,
+                                            error_message="invalid choice")
+        return validator
+
+    def _make_directory_completer(self):
+        """
+        Private method that creates a completer for a field
+        with ``data_type`` of "dir".
+
+        Returns
+        -------
+        completer : prompt_toolkit.completion.base.Completer
+            A ``Completer`` instance that suggests directory names
+            as potential completions for user input.
+        """
+        return PathCompleter(expanduser=False, only_directories=True)
+
+    def _make_directory_validator(self):
+        """
+        Private method that creates a validator for a field
+        with ``data_type`` of "dir".
+
+        Returns
+        -------
+        validator : prompt_toolkit.validation.Validator
+            A ``Validator`` instance that makes sure that only
+            directory names are chosen as the final user input.
+        """
+        validator = Validator.from_callable(lambda filepath: Path(filepath).is_dir(),
+                                            error_message="invalid directory")
+        return validator
+
+    def _make_file_completer(self):
+        """
+        Private method that creates a completer for a field
+        with ``data_type`` of "file".
+
+        Returns
+        -------
+        completer : prompt_toolkit.completion.base.Completer
+            A ``Completer`` instance that suggests directory names
+            and files with csv/tsv/xlsx extensions as potential
```

Comment:
Should we mention all extensions or maybe but an "etc." in here?

Reply:
I will do that.